### PR TITLE
Use `functools.wraps` in `track_in_mlflow` decorator

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Without `functools.wraps` decorator, `track_in_mlflow` overwrites the function name and docstring. This might be undesirable for users. Indeed,  `experimental` decorator uses this syntax as in https://github.com/optuna/optuna/blob/1a4e46c48f3fbdcf6ce5cb23f12a6659dbcae92c/optuna/_experimental.py#L57

## Description of the changes

- Add `functools.wraps` to MLFlow's `track_in_mlflow` decorator
- Add its test.

### CODE REVIEW

**Comment by hvy** [comment]
Thanks a lot for the fix, I'll take a look at it. 
@himkt could you jump in as a reviewer as well?

**Review by hvy — APPROVED** [review]
LGTM!

File: tests/integration_tests/test_mlflow.py [hunk 1]
```diff
@@ -398,8 +398,10 @@ def _objective_func(trial: optuna.trial.Trial) -> float:
         mlflow.log_metric(metric_name, metric)
         return (x - 2) ** 2 + (y - 25) ** 2 + z
 
+    tracked_objective = mlflc.track_in_mlflow()(_objective_func)
```
Comment (himkt) [inline tests/integration_tests/test_mlflow.py:401]: Please tell me why we have to stop using decorator syntax and manually invoke wrapping function?

Comment (nzw0301) [reply tests/integration_tests/test_mlflow.py:401]: To access the docstring and function name of `objective`'s `name` and `docstring`.

Comment (nzw0301) [reply tests/integration_tests/test_mlflow.py:401]: Note that decorator syntax and wrapping the function are equivalent.

**Review by himkt — APPROVED** [review]
Sorry for bothering you @nzw0301. Now LGTM.

**Comment by nzw0301** [comment]
Thanks. No worries!

**Pull request merged**